clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -249,6 +249,11 @@ public int read(byte[] b, int off, int len) throws IOException {
         LOG.warn("Recache attempt failed.", ioe);
         cancelRecache();
       }
+      // TACHYON-813: Only cache locally
+      if (mBlockOutStream instanceof RemoteBlockOutStream) {
```

Comment:
Maybe this should go right after line 247, since in this case, it is still possible that mBlockOutStream could be null at this point.